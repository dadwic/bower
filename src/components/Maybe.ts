import * as React from "react";

export interface MaybeProps {
  children: React.ReactElement;
  condition?: boolean;
}

const Maybe: React.FC<MaybeProps> = ({ children, condition }) => {
  return Boolean(condition) ? children : null;
};

export default Maybe;
