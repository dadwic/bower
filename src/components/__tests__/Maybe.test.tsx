import React from "react";
import renderer from "react-test-renderer";
import Maybe from "../Maybe";

describe("Maybe", () => {
  it("should corretly match snapshot", () => {
    const tree = renderer
      .create(
        <Maybe condition={true}>
          <span>Visible</span>
        </Maybe>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
