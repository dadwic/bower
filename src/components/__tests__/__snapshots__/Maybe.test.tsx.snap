// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maybe should corretly match snapshot 1`] = `
<span>
  Visible
</span>
`;
